frontend/public/components/resource-dropdown.tsx

```diff
@@ -92,51 +65,79 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = (props) => {
   const kinds = resources.groupBy((m) => m.kind);
   const isDup = (kind) => kinds.get(kind).size > 1;
 
-  const isKindSelected = (kind: string) => {
-    return _.includes(selected, kind);
-  };
-  // Create dropdown items for each resource.
-  const items = resources
-    .map((model) => (
-      <DropdownItem
-        key={referenceForModel(model)}
-        model={model}
-        showGroup={isDup(model.kind)}
-        checked={isKindSelected(referenceForModel(model))}
-      />
-    ))
-    .toJS();
-
-  const autocompleteFilter = (text, item) => {
-    const { model } = item.props;
-    if (!model) {
-      return false;
-    }
+  React.useEffect(() => {
+    setSelectedOptions(selected);
+  }, [selected]);
 
-    return fuzzy(
-      _.toLower(text),
-      model.labelKey ? _.toLower(t(model.labelKey)) : _.toLower(model.kind),
+  const items = [];
+  resources.forEach((model) => {
+    items.push(
+      <SelectOption
+        key={referenceForModel(model)}
+        value={`${referenceForModel(model)}`}
+        data-filter-text={`${model.abbr}${model.labelKey ? t(model.labelKey) : model.kind}`}
+      >
+        <span className={'co-resource-item'}>
+          <span className="co-resource-icon--fixed-width">
+            <ResourceIcon kind={referenceForModel(model)} />
+          </span>
+          <span className="co-resource-item__resource-name">
+            <span>
+              {model.labelKey ? t(model.labelKey) : model.kind}
+              {model.badge && model.badge === 'Tech Preview' && (
+                <span className="co-resource-item__tech-dev-preview">
+                  {t('public~Tech Preview')}
+                </span>
+              )}
+            </span>
+            {isDup(model.kind) && (
+              <span className="co-resource-item__resource-api text-muted co-truncate show co-nowrap small">
+                {model.apiGroup || 'core'}/{model.apiVersion}
+              </span>
+            )}
+          </span>
+        </span>
+      </SelectOption>,
     );
+  });
+
+  const onCustomFilter = (event) => {
```

Comment:
We should add a type for `event`. (I'm not sure the right type offhand... We should be able to check the PatternFly doc.)

Reply:
Added type using PatternFly's docs:

onFilter: event: React.ChangeEvent<HTMLInputElement>
onSelect:  React.MouseEvent | React.ChangeEvent